marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/LoadUserArtifactsCommand.java

```diff
@@ -142,15 +142,23 @@ protected Modules findModulesWithResolvedBaseDir(String resolvedBaseDir) {
                         return modules;
                     }
                 };
-                Files.walkFileTree(artifactPath, new SimpleFileVisitor<Path>() {
-                    @Override
-                    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                        if (dir.toFile().isDirectory()) {
-                            executeWalkWithDataService(dir, modulesFinder, artifactService, typeInfo);
+                if("stepDefinition".equals(artifactType)){
```

Comment:
Why does stepDefinition needs it own code here?

Reply:
All artifacts are present in base dir except stepdef, hence calling File.walk in case of stepdef,
also we don't want to reload the previously loaded mappings(using REST) using DS.